docs/docs/guides/labs/components/creating-new-component-types/creating-dg-plugin.md

```diff
@@ -0,0 +1,91 @@
+---
+title: 'Creating a `dg` plugin'
+sidebar_position: 200
+---
+
+import Preview from '@site/docs/partials/\_Preview.md';
+
+<Preview />
+
+A `dg` plugin is a Python package that defines and exposes `dg`-legible
+objects. `dg`-legible objects are collectively called _plugin objects_ and
+include custom component types, scaffolding classes, and more. Plugin objects
+that are exposed by an installed `dg` plugin can be inspected and interacted
+with by `dg` commands.
+
+Any Python package can be made into a `dg` plugin by declaring an [entry
+point](https://packaging.python.org/en/latest/specifications/entry-points/)
+under the `dagster_dg.plugin` group in its package metadata. The entry point
+indicates a module within the package that contains references to plugin
+objects.
+
+:::note
+New projects scaffolded by `dg` include a `dagster_dg.plugin` entry point,
+and therefore are `dg` plugins out of the box. This allows a project to define
+project-scoped component types etc.
+:::
+
+Let's step through the process of converting an existing third-party
+package into a `dg` plugin. We'll:
+
+- Define a custom component type in the library
+- Add a `dagster_dg.plugin` entry point to the package metadata
+- Confirm that the custom component type is available to `dg` commands
+
+Let's start with a basic package called `my_library`. `my_library` is intended
```

Comment:
Should we instead talk about building a plugin within a project? That is going to be more common than a standalone package IMO

Reply:
Ah this is what the the guide is about